src/overlays/actors/ovl_En_Snowman/z_en_snowman.h

```diff
@@ -2,16 +2,56 @@
 #define Z_EN_SNOWMAN_H
 
 #include "global.h"
+#include "objects/object_snowman/object_snowman.h"
+
+#define EN_SNOWMAN_GET_TYPE(thisx) ((thisx)->params)
+#define EN_SNOWMAN_GET_ATTACK_RANGE(thisx) (((thisx)->params >> 8) & 0xFF)
+
+typedef enum {
+    /* 0 */ EN_SNOWMAN_TYPE_SMALL,
+    /* 1 */ EN_SNOWMAN_TYPE_LARGE,
+    /* 2 */ EN_SNOWMAN_TYPE_SPLIT,
+    /* 3 */ EN_SNOWMAN_TYPE_SMALL_SNOWBALL,
+    /* 4 */ EN_SNOWMAN_TYPE_LARGE_SNOWBALL,
+} EnSnowmanType;
 
 struct EnSnowman;
 
 typedef void (*EnSnowmanActionFunc)(struct EnSnowman*, GlobalContext*);
 
 typedef struct EnSnowman {
-    /* 0x0000 */ Actor actor;
-    /* 0x0144 */ char unk_144[0x13C];
-    /* 0x0280 */ EnSnowmanActionFunc actionFunc;
-    /* 0x0284 */ char unk_284[0xF4];
+    /* 0x000 */ Actor actor;
+    /* 0x144 */ SkelAnime skelAnime;
+    /* 0x188 */ SkelAnime snowPileSkelAnime;
+    /* 0x1CC */ Vec3s jointTable[EENO_LIMB_MAX];
+    /* 0x214 */ Vec3s morphTable[EENO_LIMB_MAX];
+    /* 0x25C */ Vec3s snowPileJointTable[EENO_SNOW_PILE_LIMB_MAX];
+    /* 0x26E */ Vec3s snowPileMorphTable[EENO_SNOW_PILE_LIMB_MAX];
+    /* 0x280 */ EnSnowmanActionFunc actionFunc;
+    /* 0x284 */ EnSnowmanActionFunc oldActionFunc;
+    /* 0x288 */ u8 isHoldingSnowball;
+    /* 0x289 */ u8 combineState;
+    /* 0x28A */ u8 turningOnSteepSlope;
+    /* 0x28B */ u8 drawDmgEffType;
+    /* 0x28C */ union {
+                    s16 timer;
+                    s16 snowballsToThrowBeforeIdling;
+                };
```

Comment:
I don't think we should use anonymous unions to hold genuinely distinct things.

Reply:
I named it `work` for now, but my comment on `fwork` above still applies; I'm not sure there's a better name for something that holds two different things.

Reply:
Yeah, but my point is that you don't want it to look like "just another struct" member to someone, and at least a named union differentiates it from that. There's also the option of using defines, which is what some actors in OoT do, but I'd rather wait until we've settled what that project's going to do before making a decision here.